lib/waffle/file.ex

```diff
@@ -98,12 +98,12 @@ defmodule Waffle.File do
     string_extension =
       extension
       |> to_string()
-      |> (fn ext -> if String.starts_with?(ext, ["", "."]), do: ext, else: ".#{ext}" end).()
+      |> (fn ext -> if String.starts_with?(ext, "."), do: ext, else: ".#{ext}" end).()
 
     file_name =
       :crypto.strong_rand_bytes(20)
       |> Base.encode32()
-      |> Kernel.<>("." <> string_extension)
+      |> Kernel.<>(string_extension)
```

Comment:
It looks like we were creating a temporary path with two dots.

Reply:
Right.